tests/src/performance/perflab/DelegatePerf.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Xunit.Performance;
+using System;
+using Xunit;
+
+internal delegate long DelegateLong(Object obj, long x, long y);
+internal delegate void MultiDelegate(Object obj, long x, long y);
+
+internal delegate int SerializeDelegate();
+
+public class DelegatePerf
+{
+    [Benchmark]
+    public void DelegateInvoke()
+    {
+        DelegateLong dl = new DelegateLong(this.Invocable1);
+        Object obj = new Object();
+
+        long ret = dl(obj, 100, 100);
+
+        foreach (var iteration in Benchmark.Iterations)
+            using (iteration.StartMeasurement())
+                ret = dl(obj, 100, 100);
+    }
+
+    [Benchmark]
+    public void MulticastDelegateCombineInvoke()
+    {
+        MultiDelegate md = null;
+        Object obj = new Object();
+
+        foreach (var iteration in Benchmark.Iterations)
+        {
+            MultiDelegate md1 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md2 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md3 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md4 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md5 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md6 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md7 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md8 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md9 = new MultiDelegate(this.Invocable2);
+            MultiDelegate md10 = new MultiDelegate(this.Invocable2);
```

Comment:
Do we need to recreate these delegates on each iteration, or could these be moved out of the loop?